### DISCUSSION TITLE
Pandas group-by filter outputs to Awkward

### DISCUSSION THREAD

**Discussion by gipert** [scikit-hep/awkward#2755] [Q&A]
Assume that a filter operation is performed on a column after a `pandas.DataFrame.groupby()` call, and this results in a variable number of elements (of that column) in each obtained group. Would it be possible to interpret such a Series as an Awkward array?

Here's for example a dataframe that contains variably sized groups, after grouping on `event`:

```pycon
>>> import pandas as pd
>>> df = pd.DataFrame({
>>>   "event": [0, 0, 1, 2, 2, 2],
>>>   "ch_energy": [0.1, 0.2, 0.5, 0.8, 1.2, 4.2
>>> })
>>> df.groupby("event").ch_energy.size()
event
1    2
2    1
3    3
```

I would be interested into extracting an awkward array that looks like this:
```text
[[0.1, 0.2],
 [0.5],
 [0.8, 1.2, 4.2]
]
```

Thanks for helping!

**Comment by agoose77** [comment]
I can certainly weigh in here, but a MWE would be useful to frame our discussion!

**Reply by gipert** [reply]
Hi @agoose77, I have edited the original post!

**Reply by agoose77** [reply]
@douglasdavis I anticipated that there might be a straightforward function in `awkward-pandas` to leverage the result of a `groupby`. Is that not the case?
